rgd/geodata/models/fmv/etl.py

```diff
@@ -17,59 +16,44 @@
 logger = get_task_logger(__name__)
 
 
-def _extract_klv_with_docker(fmv_file_entry):
-    logger.info('Entered `_extract_klv_with_docker`')
-    image_name = 'banesullivan/kwiver:dump-klv'
+def _extract_klv(fmv_file_entry):
+    logger.info('Entered `_extract_klv`')
+
+    if not shutil.which('kwiver'):
+        raise RuntimeError('kwiver not installed. Run `pip install kwiver`.')
+
     video_file = fmv_file_entry.file
-    try:
-        client = docker.from_env(version='auto', timeout=3600)
-        _ = client.images.pull(image_name)
-
-        with video_file.yield_local_path() as dataset_path:
-            logger.info('Running dump-klv with data %s' % (dataset_path))
-            tmpdir = tempfile.mkdtemp()
-            output_path = os.path.join(tmpdir, os.path.basename(video_file.file.name) + '.klv')
-            stderr_path = os.path.join(tmpdir, 'stderr.dat')
-            cmd = [
-                'docker',
-                'run',
-                '--rm',
-                '-i',
-                image_name,
-            ]
-            logger.info(
-                'Running %s <%s >%s 2>%s'
-                % (
-                    ' '.join([shlex.quote(c) for c in cmd]),
-                    shlex.quote(str(dataset_path)),
-                    shlex.quote(output_path),
-                    shlex.quote(stderr_path),
-                )
-            )
-            try:
-                subprocess.check_call(
-                    cmd,
-                    stdin=open(dataset_path, 'rb'),
-                    stdout=open(output_path, 'wb'),
-                    stderr=open(stderr_path, 'wb'),
-                )
-                result = 0
-            except subprocess.CalledProcessError as exc:
-                result = exc.returncode
-                logger.info('Failed to successfully run image (%r)' % (exc))
-                raise exc
-            logger.info('Finished running image with result %r' % result)
-            # Store result
-            fmv_file_entry.klv_file.save(os.path.basename(output_path), open(output_path, 'rb'))
-            fmv_file_entry.save(
-                update_fields=[
-                    'klv_file',
-                ]
+    with video_file.yield_local_path() as dataset_path:
+        logger.info('Running dump-klv with data %s' % (dataset_path))
+        tmpdir = tempfile.mkdtemp()
```

Comment:
Let's use the [context manager form of this, TemporaryDirectory](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory). That will obviate the need to call `shutil.rmtree` later.